### ISSUE TITLE
Analytics > Products > Show Single Product > Single variation - 400 error

### ISSUE THREAD

**Issue by AlchemyUnited** [woothemes/woocommerce#45911]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When using the Admin > Analytics > Products > Show Single Product > Single variation - the first two sections on the page in the Report area render:

![image](https://github.com/woocommerce/woocommerce/assets/1966613/f0e3d202-e7d3-4793-ad9c-a84cadcc0c60)

In the browser Console it shows the request URL pulls a 400. Actually, it's 2 URLs w/ a 400, I presume one for each section. Here's a sample of one of them:

https://MY-SITE.com/wp-json/wc-analytics/reports/products/stats?order=asc&interval=day&per_page=100&after=2024-03-01T00%3A00%3A00&before=2024-03-25T23%3A59%3A59&segmentby=variation&fields%5B0%5D=items_sold&fields%5B1%5D=net_revenue&fields%5B2%5D=orders_count&products=MY-PRODUCT-ID&_locale=user

After doing some additional digging in the Console the response in full is:

{"code":"wc_admin_reports_invalid_segmenting_variation",
"message":"product_includes parameter need to specify exactly one product when segmenting by variation.",
"data":{"status":400}
}

For easy of use I slapped the same URL in Postman... Well, there is no  parameter product_includes in the request. That is the error indicates something is missing. 

From there I looked into WP core and searched for: ""product_includes parameter" and found: 

src / Admin / API / Reports / Products / Stats / Segmentor.php > Starting at line 153

```
// Product, variation, and category are bound to product, so here product segmenting table is required,
// while coupon and customer are bound to order, so we don't need the extra JOIN for those.
// This also means that segment selections need to be calculated differently.
		if ( 'product' === $this->query_args['segmentby'] ) {
			$product_level_columns     = $this->get_segment_selections_product_level( $product_segmenting_table );
			$segmenting_selections     = array(
				'product_level' => $this->prepare_selections( $product_level_columns ),
			);
			$this->report_columns      = $product_level_columns;
			$segmenting_from           = '';
			$segmenting_groupby        = $product_segmenting_table . '.product_id';
			$segmenting_dimension_name = 'product_id';

			$segments = $this->get_product_related_segments( $type, $segmenting_selections, $segmenting_from, $segmenting_where, $segmenting_groupby, $segmenting_dimension_name, $table_name, $query_params, $unique_orders_table );
		} elseif ( 'variation' === $this->query_args['segmentby'] ) {
			if ( ! isset( $this->query_args['product_includes'] ) || count( $this->query_args['product_includes'] ) !== 1 ) {
				throw new ParameterException( 'wc_admin_reports_invalid_segmenting_variation', __( 'product_includes parameter need to specify exactly one product when segmenting by variation.', 'woocommerce' ) );
			}
```
Back to Postman...

Changed the URL / Request so segementby is product, instead of variation. That gets results. I can't exactly tell if it's correct since it's JSON. But at least it's not a 400. 

### Expected behavior

The report shows data / info, not "There was an error getting your stats. Please try again."

### Actual behavior

See above

### Steps to reproduce

Note: Unfortuntely, ATM, there's only a production env, and therefore can deactivate all the plugins, etc. That said, it appears to be an issue with the FE JS and query / URL it build. I suppose a plugin could be conflicting but that seems unlikely in this case, yes? 

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): https://*****.com
Site address (URL): https://*****.com
WC Version: 8.7.0
REST API Version: ✔ 8.7.0
Action Scheduler Version: ✔ 3.7.2
Log Directory Writable: ✔
WP Version: 6.4.3
WP Multisite: –
WP Memory Limit: 1 GB
WP Debug Mode: –
WP Cron: –
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 8.2.14
PHP Post Max Size: 100 MB
PHP Time Limit: 43300
PHP Max Input Vars: 10000
cURL Version: 7.81.0
OpenSSL/3.0.2

SUHOSIN Installed: –
MySQL Version: 8.0.36-28
Max Upload Size: 50 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 8.7.0
WC Database Prefix: wp_
Total Database Size: 4909.47MB
Database Data Size: 2864.72MB
Database Index Size: 2044.75MB
wp_woocommerce_sessions: Data: 81.02MB + Index: 0.13MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 53.59MB + Index: 20.56MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 538.00MB + Index: 449.98MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 1.52MB + Index: 1.19MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 1.52MB + Index: 3.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 1.52MB + Index: 0.39MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 4.52MB + Index: 5.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 38.06MB + Index: 36.17MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 29.05MB + Index: 20.03MB + Engine InnoDB
wp_apexnb_subscriber: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_atum_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_atum_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_atum_product_data: Data: 0.25MB + Index: 0.20MB + Engine InnoDB
wp_auctions_for_woocommerce_log: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_bwf_action_claim: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_bwf_actions: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_bwf_contact: Data: 10.52MB + Index: 12.06MB + Engine InnoDB
wp_bwf_contact_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_bwf_conversion_tracking: Data: 0.02MB + Index: 0.33MB + Engine InnoDB
wp_bwf_wc_customers: Data: 8.52MB + Index: 1.86MB + Engine InnoDB
wp_commentmeta: Data: 7.52MB + Index: 8.03MB + Engine InnoDB
wp_comments: Data: 364.92MB + Index: 206.52MB + Engine InnoDB
wp_gglcptch_allowlist: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_imagify_files: Data: 0.05MB + Index: 0.08MB + Engine InnoDB
wp_imagify_folders: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_jetpack_sync_queue: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 6.44MB + Index: 0.14MB + Engine InnoDB
wp_postmeta: Data: 918.00MB + Index: 971.97MB + Engine InnoDB
wp_posts: Data: 72.47MB + Index: 43.11MB + Engine InnoDB
wp_psmsc_agents: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_psmsc_attachments: Data: 0.14MB + Index: 0.00MB + Engine InnoDB
wp_psmsc_canned_reply: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_psmsc_categories: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_psmsc_cr_categories: Data: 1.52MB + Index: 0.00MB + Engine InnoDB
wp_psmsc_custom_fields: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_psmsc_customers: Data: 1.52MB + Index: 0.00MB + Engine InnoDB
wp_psmsc_email_notifications: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_psmsc_email_otp: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_psmsc_holidays: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_psmsc_logs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_psmsc_options: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_psmsc_priorities: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_psmsc_scheduled_tasks: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_psmsc_statuses: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_psmsc_threads: Data: 23.52MB + Index: 0.00MB + Engine InnoDB
wp_psmsc_ticket_tags: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_psmsc_tickets: Data: 1.52MB + Index: 0.00MB + Engine InnoDB
wp_psmsc_wh_exceptions: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_psmsc_working_hrs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_salesmatelogs: Data: 0.11MB + Index: 0.00MB + Engine InnoDB
wp_say_what_strings: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_taxjar_record_queue: Data: 12.52MB + Index: 5.52MB + Engine InnoDB
wp_term_relationships: Data: 0.31MB + Index: 0.25MB + Engine InnoDB
wp_term_taxonomy: Data: 0.08MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.08MB + Index: 0.09MB + Engine InnoDB
wp_usermeta: Data: 149.70MB + Index: 136.39MB + Engine InnoDB
wp_users: Data: 10.52MB + Index: 8.55MB + Engine InnoDB
wp_vtprd_purchase_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_vtprd_purchase_log_product: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_vtprd_purchase_log_product_rule: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 7.52MB + Index: 8.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_addresses: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 2.52MB + Index: 4.03MB + Engine InnoDB
wp_wc_order_operational_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 34.56MB + Index: 38.09MB + Engine InnoDB
wp_wc_order_stats: Data: 18.55MB + Index: 13.55MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 5.52MB + Index: 6.03MB + Engine InnoDB
wp_wc_orders: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_wc_orders_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.22MB + Index: 0.42MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.08MB + Index: 0.05MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.45MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wcr_coupon_verification: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wdr_order_discounts: Data: 1.52MB + Index: 0.00MB + Engine InnoDB
wp_wdr_order_item_discounts: Data: 1.52MB + Index: 0.41MB + Engine InnoDB
wp_wdr_rules: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfblockediplog: Data: 0.09MB + Index: 0.00MB + Engine InnoDB
wp_wfblocks7: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wfco_report_views: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wfconfig: Data: 4.50MB + Index: 0.00MB + Engine InnoDB
wp_wfcrawlers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wffilechanges: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wffilemods: Data: 13.55MB + Index: 0.00MB + Engine InnoDB
wp_wfhits: Data: 2.02MB + Index: 0.22MB + Engine InnoDB
wp_wfhoover: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfissues: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wfknownfilelist: Data: 6.52MB + Index: 0.00MB + Engine InnoDB
wp_wflivetraffichuman: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wflocs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wflogins: Data: 1.02MB + Index: 0.19MB + Engine InnoDB
wp_wfls_2fa_secrets: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfls_role_counts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfls_settings: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfnotifications: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_wfob_stats: Data: 23.56MB + Index: 34.06MB + Engine InnoDB
wp_wfocu_event: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wfocu_event_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfocu_session: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wfpendingissues: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wfreversecache: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfsecurityevents: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfsnipcache: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wfstatus: Data: 0.27MB + Index: 0.16MB + Engine InnoDB
wp_wftrafficrates: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_wfwafconfig: Data: 2.52MB + Index: 0.00MB + Engine InnoDB
wp_wfwaffailures: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_square_customers: Data: 3.52MB + Index: 0.00MB + Engine InnoDB
wp_wpf_logging: Data: 15.05MB + Index: 0.19MB + Engine InnoDB
wp_wpforms_tasks_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpie_template: Data: 0.27MB + Index: 0.00MB + Engine InnoDB
wp_wpmailsmtp_debug_events: Data: 1.52MB + Index: 0.00MB + Engine InnoDB
wp_wpmailsmtp_emails_log: Data: 373.56MB + Index: 3.86MB + Engine InnoDB
wp_wpmailsmtp_tasks_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpr_rocket_cache: Data: 0.13MB + Index: 0.20MB + Engine InnoDB
wp_wpr_rucss_used_css: Data: 0.33MB + Index: 0.44MB + Engine InnoDB
wp_wt_iew_action_history: Data: 0.09MB + Index: 0.00MB + Engine InnoDB
wp_wt_iew_cron: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wt_iew_ftp: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wt_iew_mapping_template: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_yoast_indexable: Data: 1.52MB + Index: 0.44MB + Engine InnoDB
wp_yoast_indexable_hierarchy: Data: 0.08MB + Index: 0.08MB + Engine InnoDB
wp_yoast_migrations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yoast_primary_term: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_yoast_prominent_words: Data: 0.42MB + Index: 0.78MB + Engine InnoDB
wp_yoast_seo_links: Data: 0.44MB + Index: 0.45MB + Engine InnoDB
wp_yoast_seo_meta: Data: 0.41MB + Index: 0.00MB + Engine InnoDB

### Post Type Counts ###

attachment: 2765
custom_css: 2
customize_changeset: 3
flamingo_contact: 516
flamingo_inbound: 366
jetpack_migration: 2
nav_menu_item: 78
o-list: 8
oembed_cache: 11
omapi: 6
page: 70
post: 116
product: 360
product_variation: 1296
revision: 153
sa_slider: 2
shop_coupon: 76605
shop_order: 154796
shop_order_refund: 1275
vc_grid_item: 1
wcct_countdown: 1
wcs_ruleset: 8
wfacp_checkout: 2
wfob_bump: 16
wfocu_funnel: 3
wfocu_offer: 6
wpb_wtm_tab: 1
wpcf7_contact_form: 4
wpf_cart: 2263
wpforms: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (53) ###

Age Verification for eCommerce: by AgeChecker.Net – 1.16.2
ATUM Inventory Management for WooCommerce: by Stock Management Labs™ – 1.9.37.2
WP Engine Smart Plugin Manager: by WP Engine – 5.17.50
PublishPress Capabilities Pro: by PublishPress – 2.10.1
Salesmate Integration for Contact Form 7: by Salesmate io. – 1.7
Coinbase Commerce: by Coinbase Commerce – 1.3
Contact Form 7: by Takayuki Miyoshi – 5.9.3
Yoast Duplicate Post: by Enrico Battocchi & Team Yoast – 4.5
GTM4WP - A Google Tag Manager (GTM) plugin for WordPress: by Thomas Geiger – 1.20.1
Email Marketing by Drip: by Drip – 1.0.2
Envato Market: by Envato – 2.0.11
Hide Categories and Products for Woocommerce: by N.O.U.S. Open Useful and Simple – 1.2.9
Imagify: by Imagify – Optimize Images & Convert WebP & Avif – 2.2.0.1
Jetpack: by Automattic – 13.2.2
WPBakery Page Builder: by Michael M - WPBakery.com – 7.5
Live Cart by WP Engine: by WP Engine – 1.0.1
Savoy Theme - Content Elements: by NordicMade – 1.5.2
Savoy Theme - Settings Panel: by NordicMade – 1.0.9
OptinMonster: by OptinMonster Popup Builder Team – 2.16.0
Post Affiliate Pro: by QualityUnit – 1.26.10
Route App: by Route – 2.2.10
Sezzle WooCommerce Payment: by Sezzle – 5.0.8
Slide Anything PRO - Modal Popups for Slide Anything: by Simon Edge – 2.0.1
Slide Anything - Responsive Content / HTML Slider and Carousel: by  – 2.4.9
SupportCandy: by SupportCandy – 3.2.6
TaxJar - Sales Tax Automation for WooCommerce: by TaxJar – 4.2.1
Orbit Fox Companion: by Themeisle – 2.10.34
WooCommerce Conditional Shipping Pro: by Lauri Karisola / WP Trio – 3.2.0
Woo Discount Rules PRO 2.0: by Flycart – 2.6.1
Discount Rules Core: by Flycart – 2.6.3
WooCommerce Checkout Field Editor: by WooCommerce – 1.7.12
WooCommerce Coupon Restrictions: by WooCommerce – 2.2.2
WooCommerce - ShipStation Integration: by WooCommerce – 4.4.0 (update to version 4.4.1 is available)
WooCommerce Square: by WooCommerce – 4.5.1
WooCommerce: by Automattic – 8.7.0
FunnelKit Order Bumps: by Funnelkit – 1.20.2
Wordfence Security: by Wordfence – 7.11.4
Yoast SEO Premium: by Team Yoast – 22.3
Yoast SEO: by Team Yoast – 22.3
WP Fusion - Abandoned Cart Addon: by Very Good Plugins – 1.8.0
WP Fusion - Enhanced Ecommerce Addon: by Very Good Plugins – 1.23.5
WP Fusion: by Very Good Plugins – 3.43.1.1
WP Mail SMTP Pro: by WP Mail SMTP – 4.0.2
WP Rocket: by WP Media – 3.15.10
SupportCandy - Assign Agent Rules: by SupportCandy – 3.0.5
SupportCandy - Automatic close tickets: by SupportCandy – 3.0.5
SupportCandy - Canned Reply: by SupportCandy – 3.0.6
SupportCandy - Email Piping: by SupportCandy – 3.1.8
SupportCandy - Private Credentials: by SupportCandy – 3.0.5
SupportCandy - WooCommerce: by SupportCandy – 3.1.2
Yoast SEO: WooCommerce: by Team Yoast – 16.1.1
Yotpo Social Reviews for Woocommerce: by Yotpo – 1.7.8

### Inactive Plugins (2) ###

Query Monitor: by John Blackbourn – 3.15.0
WP Import Export: by VJInfotech – 3.9.27

### Dropin Plugins () ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached Redux

### Must Use Plugins (6) ###

Arete MU Plugins: by Arete – 1.0.0
Force Strong Passwords - WPE Edition: by Jason Cosper – 1.8.0
WP Engine Cache Plugin: by WP Engine – 1.3.0
WP Engine Seamless Login Plugin: by WP Engine – 1.6.1
WP Engine Security Auditor: by wpengine – 1.1.1
WP Engine System: by WP Engine – 5.0.1

### Settings ###

API Enabled: –
Force SSL: ✔
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: auction (auction)
external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to Woo.com: ✔
Enforce Approved Product Download Directories: –
HPOS feature screen enabled: –
HPOS feature enabled: –
Order datastore: WC_Order_Data_Store_CPT
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 178 MB

### WC Pages ###

Shop base: #85071 - /cbd-shop/
Cart: #824 - /cart/
Checkout: #761 - /checkout/
My account: #825 - /my-account-2/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Savoy Child
Version: 1.0.0
Author URL: 
Child Theme: ✔
Parent Theme Name: Savoy
Parent Theme Version: 2.9.2
Parent Theme Author URL: http://www.nordicmade.com
WooCommerce Support: ✔

### Templates ###

Overrides: savoy/woocommerce/archive-product.php
savoy/woocommerce/cart/cart-empty.php
savoy/woocommerce/cart/cart-item-data.php
savoy/woocommerce/cart/cart-shipping.php
savoy/woocommerce/cart/cart-totals.php
savoy/woocommerce/cart/cart.php
savoy/woocommerce/cart/cross-sells.php
savoy/woocommerce/cart/mini-cart.php
savoy/woocommerce/cart/proceed-to-checkout-button.php
savoy/woocommerce/cart/shipping-calculator.php
savoy/woocommerce/checkout/cart-errors.php
savoy/woocommerce/checkout/form-billing.php
savoy/woocommerce/checkout/form-checkout.php
savoy/woocommerce/checkout/form-coupon.php
savoy/woocommerce/checkout/form-login.php
savoy/woocommerce/checkout/form-pay.php
savoy/woocommerce/checkout/form-shipping.php
savoy/woocommerce/checkout/order-received.php
savoy/woocommerce/checkout/payment-method.php
savoy/woocommerce/checkout/payment.php
savoy/woocommerce/checkout/review-order.php
savoy/woocommerce/checkout/terms.php
savoy/woocommerce/checkout/thankyou.php
savoy/woocommerce/content-product.php
savoy/woocommerce/content-single-product.php
savoy/woocommerce/content-widget-price-filter.php
savoy/woocommerce/content-widget-product.php
savoy/woocommerce/content-widget-reviews.php
savoy/woocommerce/global/breadcrumb.php
savoy/woocommerce/global/form-login.php
savoy/woocommerce/global/quantity-input.php
savoy/woocommerce/global/sidebar.php
savoy/woocommerce/global/wrapper-end.php
savoy/woocommerce/global/wrapper-start.php
savoy/woocommerce/loop/add-to-cart.php
savoy/woocommerce/loop/loop-end.php
savoy/woocommerce/loop/loop-start.php
savoy/woocommerce/loop/no-products-found.php
savoy/woocommerce/loop/pagination.php
savoy/woocommerce/loop/price.php
savoy/woocommerce/loop/rating.php
savoy/woocommerce/loop/sale-flash.php
savoy/woocommerce/myaccount/dashboard.php
savoy/woocommerce/myaccount/downloads.php
savoy/woocommerce/myaccount/form-add-payment-method.php
savoy/woocommerce/myaccount/form-edit-account.php
savoy/woocommerce/myaccount/form-edit-address.php
savoy/woocommerce/myaccount/form-login.php
savoy/woocommerce/myaccount/form-lost-password.php
savoy/woocommerce/myaccount/form-reset-password.php
savoy/woocommerce/myaccount/lost-password-confirmation.php
savoy/woocommerce/myaccount/my-account.php
savoy/woocommerce/myaccount/my-address.php
savoy/woocommerce/myaccount/navigation.php
savoy/woocommerce/myaccount/orders.php
savoy/woocommerce/myaccount/payment-methods.php
savoy/woocommerce/myaccount/view-order.php
savoy/woocommerce/notices/error.php
savoy/woocommerce/notices/notice.php
savoy/woocommerce/notices/success.php
savoy/woocommerce/order/form-tracking.php
savoy/woocommerce/order/order-again.php
savoy/woocommerce/order/order-details-customer.php
savoy/woocommerce/order/order-details-item.php
savoy/woocommerce/order/order-details.php
savoy/woocommerce/order/tracking.php
savoy/woocommerce/product-searchform.php
savoy/woocommerce/single-product/add-to-cart/external.php
savoy/woocommerce/single-product/add-to-cart/grouped.php
savoy/woocommerce/single-product/add-to-cart/simple.php
savoy/woocommerce/single-product/add-to-cart/variable.php
savoy/woocommerce/single-product/add-to-cart/variation-add-to-cart-button.php
savoy/woocommerce/single-product/add-to-cart/variation.php
savoy/woocommerce/single-product/meta.php
savoy/woocommerce/single-product/photoswipe.php
savoy/woocommerce/single-product/price.php
savoy/woocommerce/single-product/product-attributes.php
savoy/woocommerce/single-product/product-image.php
savoy/woocommerce/single-product/product-thumbnails.php
savoy/woocommerce/single-product/rating.php
savoy/woocommerce/single-product/related.php
savoy/woocommerce/single-product/review-meta.php
savoy/woocommerce/single-product/review-rating.php
savoy/woocommerce/single-product/review.php
savoy/woocommerce/single-product/sale-flash.php
savoy/woocommerce/single-product/share.php
savoy/woocommerce/single-product/short-description.php
savoy/woocommerce/single-product/stock.php
savoy/woocommerce/single-product/tabs/additional-information.php
savoy/woocommerce/single-product/tabs/description.php
savoy/woocommerce/single-product/tabs/tabs.php
savoy/woocommerce/single-product/title.php
savoy/woocommerce/single-product/up-sells.php
savoy/woocommerce/single-product-reviews.php
savoy/woocommerce/single-product.php
savoy/woocommerce/taxonomy-product-cat.php
savoy/woocommerce/taxonomy-product-tag.php
savoy/woocommerce/content-product_cat.php

### Square ###

Environment: Production
Tokenization Enabled: ✔
Debug Mode: Off

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-variation-management
product-virtual-downloadable
product-external-affiliate
product-grouped
product-linked
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: customize-store
minified-js
new-product-management-experience
product-pre-publish-modal
settings
async-product-editor-category-field

Daily Cron: ✔ Next scheduled: 2024-03-26 00:15:40 -04:00
Options: ✔
Notes: 52
Onboarding: completed

### Action Scheduler ###

Complete: 98,927
Oldest: 2024-02-23 19:02:20 -0500
Newest: 2024-03-25 18:09:38 -0400

Failed: 4
Oldest: 2024-03-08 08:11:08 -0500
Newest: 2024-03-22 08:54:23 -0400

Pending: 11
Oldest: 2024-03-25 18:19:03 -0400
Newest: 2024-04-01 15:34:09 -0400

### Status report information ###

Generated at: 2024-03-25 18:10:02 -04:00
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by AlchemyUnited** [comment]
Update - The client got me access to another env, staging. 

Cloned the current prod site via the hosting's control panel, deactivated ALL plugins sans WooComm, changed the theme to Twenty Twenty. Analytics > Products... Still not working :(  

I also learned that there's a dev env. But the error above does NOT happen. That is, Analytics > Products works as expected. I'm told that the env has HPOS enabled. Other than that, afaik, the envs are all identical. 

I'm going to update the new staging to HPOS and see if that changes anything.

**Comment by AlchemyUnited** [comment]
another update... The dev site was running on a slightly older version of WC. That env's Analytics > Products was working. Then I updated WC and the DB update ran and... BOOM!  Analytics > Products in dev is now busted as well.

**Comment by Stojdza** [comment]
Hello @AlchemyUnited 👋 

The fix for this issue has been included in [the WooCommerce 8.8 RC release](https://developer.woo.com/2024/03/26/woocommerce-8-8/#h-release-candidate), which is out today. If you’d like to test it out and give us feedback, you can download the pre-release package [here](https://downloads.wordpress.org/plugin/woocommerce.8.8.0-rc.1.zip).

Thanks for reporting this – your help makes WooCommerce better for everyone!

**Comment by AlchemyUnited** [comment]
@Stojdza  - Good work! And thanks for letting me know.
